File: sdk/browser/nvsodide.js
/**
 * NVsodide JS Bridge
 * Allows browser-side applications to interact with Nλvescript components.
 */
class NVsodide {
    constructor() {
        this.instance = null;
    }

    async init(wasmUrl, policies = []) {
        this.policies = policies; // e.g., ["fs:read", "net:connect"]
        const response = await fetch(wasmUrl);
        const bytes = await response.arrayBuffer();
        
        const obj = await WebAssembly.instantiate(bytes, {
            nasi: {
                log: (msg) => console.log(msg),
                syscall: (id, args) => {
                    const cap = this.mapSyscallToCap(id);
                    if (!this.policies.includes(cap)) {
                        throw new Error(`Permission Denied: Capability '${cap}' not granted.`);
                    }
                    console.log(`⚡ Browser NASI syscall: ${cap}`);
                    return 0;
                }
            }
        });
        this.instance = obj.instance;
    }

    mapSyscallToCap(id) {
        const caps = { 1: "fs:read", 2: "fs:write", 3: "net:connect" };
        return caps[id] || "unknown";
    }


    run(modulePath) {
        return this.instance.exports.nv_run(modulePath);
    }
}

File: sdk/browser/nvsodide.rs
// NVsodide: Browser Runtime Bridge
// Exposes Nλvescript execution and NASI capabilities to JS

#import <wasmtime.h>

@wasm_export("nv_init")
pub fn nv_init() -> i32 {
    // Initialize Nλvescript Runtime for Browser
    0
}

@wasm_export("nv_run")
pub fn nv_run(module_ptr: *const i8) -> i32 {
    // Run Nλvescript NASM module in browser
    0
}

@wasm_export("nv_nasi_call")
pub fn nv_nasi_call(syscall: i32, args: *const i8) -> i32 {
    // Map browser-specific APIs (DOM, fetch, storage) to NASI
    0
}
